zebra-state/src/service.rs

```diff
@@ -610,6 +576,8 @@ impl Service<Request> for StateService {
     #[instrument(name = "state", skip(self, req))]
     fn call(&mut self, req: Request) -> Self::Future {
```

Comment:
Not sure if the Requests that are now redirected to the ReadStateService should be counted as both state service and read_state, but might be nicer to have it in a `.count_metric()` method than a `.name()` method

```suggestion
        metrics::counter!(
            "state.requests",
            1,
            "service" => "state",
            "type" => match req {
                Request::Depth(_) => "depth",
                Request::Tip => "tip",
                Request::BlockLocator => "block_locator",
                Request::Transaction(_) => "transaction",
                Request::Block(_) => "block",
                Request::FindBlockHashes { .. } => "find_block_hashes",
                Request::FindBlockHeaders { .. } => "find_block_headers",
                Request::CommitBlock(_) => "commit_block",
                Request::CommitFinalizedBlock(_) => "commit_finalized_block",
                Request::AwaitUtxo(_) => "await_utxo",
            }
        );

```

Reply:
These changes can also go in that separate PR.

Reply:
> Not sure if the Requests that are now redirected to the ReadStateService should be counted as both state service and read_state

I think it's ok for now, because some requests (`Block` and `Transaction`) are used via both `StateService` and `ReadStateService` directly. So their metrics will be different across the services.

Eventually I'd like to use `ReadStateService` directly, get rid of the read requests in the `StateService`, and then the corresponding metrics will go as well. But that's something for a future cleanup.